src/immutable_data.rs

```diff
@@ -121,26 +124,17 @@ mod tests {
         let owner1 = SecretKey::random().public_key();
         let owner2 = SecretKey::random().public_key();
 
-        let idata1 = UnpubImmutableData {
-            data: data1.clone(),
-            owners: owner1,
-        };
-        let idata2 = UnpubImmutableData {
-            data: data1,
-            owners: owner2,
-        };
-        let idata3 = UnpubImmutableData {
-            data: data2,
-            owners: owner1,
-        };
-
-        assert_eq!(idata1.name(), idata1.name());
-        assert_eq!(idata2.name(), idata2.name());
-        assert_eq!(idata3.name(), idata3.name());
-
-        assert_ne!(idata1.name(), idata2.name());
-        assert_ne!(idata1.name(), idata3.name());
-        assert_ne!(idata2.name(), idata3.name());
+        let idata1 = UnpubImmutableData::new(data1.clone(), owner1);
+        let idata2 = UnpubImmutableData::new(data1, owner2);
+        let idata3 = UnpubImmutableData::new(data2, owner1);
+
+        assert_eq!(idata1.name, idata1.name);
```

Comment:
This would test for `Eq`/`PartialEq` implementation of `XorName`, so I think we shouldn't really need this?
The `assert_ne!` block should be enough.